### ISSUE TITLE
Repair broken JSON by counting delimiters.

### ISSUE THREAD

**Issue by gloryknight** [josdejong/jsonrepair#78]
It would be nice if the project could close the brackets at the end of the JSON similar to https://github.com/half-pie/half-json .
This could be done by counting opening and closing brackets+quotes and adding proper brackets at the end of the file.

**Comment by josdejong** [comment]
Thanks for your input. The library already does so: input like `{ "items": [1, 2, 3` is turned into output `{ "items": [1, 2, 3]}`

Is that what you mean?

**Comment by gloryknight** [comment]
Thank you for the clarification.
It looks like the library does not add brackets at the end of a string.
On this example `{"meta": {"na` a message `End of string expected (char 10)` is shown but no valid json output.
I do understand that closing the string will most likely corrupt the input data, but most of the actions performed by this library are likely to corrupt the data. The aim is to fix (recover) as much as it is possible.
I would suggest to add automatic closing brackets when needed. Also adding a `: null` for not completed keys would fix the JSON.

**Comment by josdejong** [comment]
Would be really nice to also fix broken strings at the end, or a missing value. Good idea!

**Comment by josdejong** [comment]
I had some more thought on this. When a JSON string ends with a half string, the text (and the JSON) is most probably not complete. It is on a different level then removing a trailing comma or replacing single quotes with double quotes. Maybe we can somehow distinguish between that, and get feedback on whether their where repairs like this one involving invalid data, or just "cosmetic" repairs.

**Comment by KevinDanikowski** [comment]
@josdejong  I feel like an "extremeFix"|"autoComplete" setting or something could be useful here. I agree with your thoughts on it being incomplete json, but perhaps the json they needed was in the completed part of the json, or they just want what they can get.

PS: Great package ! Really appreciate this, solved a big bug I was having

**Comment by josdejong** [comment]
Yes, exactly Kevin, I'm thinking in that kind of direction. Thanks for your inputs 😄

**Comment by josdejong** [comment]
> I do understand that closing the string will most likely corrupt the input data, but most of the actions performed by this library are likely to corrupt the data. The aim is to fix (recover) as much as it is possible.

I changed my mind on this a bit, I think it is more useful to have a truncated JSON repaired than nothing. I've now fixed this via 17a002a55c6f0fdeb6bf064d85d4ff6b03509963 (it was already partly working).

I want to give better control over what kind of fixes the library applies though, and insight in what fixes have been applied. That is discussed in #77 and #79.
